packages/block-editor/src/components/use-on-block-drop/index.js

```diff
@@ -68,138 +70,129 @@ export function onBlockDrop(
+		clearSelectedBlock();
+		insertBlocks(
+			blocks,
+			targetBlockIndex,
+			targetRootClientId,
+			true,
+			null
+		);
+	}
 
-		// If the user is moving a block
-		if ( dropType === 'block' ) {
-			const sourceBlockIndex = getBlockIndex(
-				sourceClientIds[ 0 ],
-				sourceRootClientId
-			);
+	// If the user is moving a block
+	if ( dropType === 'block' ) {
+		const sourceBlockIndex = getBlockIndex(
+			sourceClientIds[ 0 ],
+			sourceRootClientId
+		);
 
-			// If the user is dropping to the same position, return early.
-			if (
-				sourceRootClientId === targetRootClientId &&
-				sourceBlockIndex === targetBlockIndex
-			) {
-				return;
-			}
+		// If the user is dropping to the same position, return early.
+		if (
+			sourceRootClientId === targetRootClientId &&
+			sourceBlockIndex === targetBlockIndex
+		) {
+			return;
+		}
 
-			// If the user is attempting to drop a block within its own
-			// nested blocks, return early as this would create infinite
-			// recursion.
-			if (
-				sourceClientIds.includes( targetRootClientId ) ||
-				getClientIdsOfDescendants( sourceClientIds ).some(
-					( id ) => id === targetRootClientId
-				)
-			) {
-				return;
-			}
+		// If the user is attempting to drop a block within its own
+		// nested blocks, return early as this would create infinite
+		// recursion.
+		if (
+			sourceClientIds.includes( targetRootClientId ) ||
+			getClientIdsOfDescendants( sourceClientIds ).some(
+				( id ) => id === targetRootClientId
+			)
+		) {
+			return;
+		}
 
-			const isAtSameLevel = sourceRootClientId === targetRootClientId;
-			const draggedBlockCount = sourceClientIds.length;
+		const isAtSameLevel = sourceRootClientId === targetRootClientId;
+		const draggedBlockCount = sourceClientIds.length;
 
-			// If the block is kept at the same level and moved downwards,
-			// subtract to take into account that the blocks being dragged
-			// were removed from the block list above the insertion point.
-			const insertIndex =
-				isAtSameLevel && sourceBlockIndex < targetBlockIndex
-					? targetBlockIndex - draggedBlockCount
-					: targetBlockIndex;
+		// If the block is kept at the same level and moved downwards,
+		// subtract to take into account that the blocks being dragged
+		// were removed from the block list above the insertion point.
+		const insertIndex =
+			isAtSameLevel && sourceBlockIndex < targetBlockIndex
+				? targetBlockIndex - draggedBlockCount
+				: targetBlockIndex;
 
-			moveBlocksToPosition(
-				sourceClientIds,
-				sourceRootClientId,
-				targetRootClientId,
-				insertIndex
-			);
-		}
-	};
+		moveBlocksToPosition(
+			sourceClientIds,
+			sourceRootClientId,
+			targetRootClientId,
+			insertIndex
+		);
+	}
 }
 
 /**
- * A function that returns an event handler function for block-related file drop events.
+ * An event handler function for block-related file drop events.
  *
+ * @param {Array}    files                 The files array from the drop event.
```

Comment:
Here the same - if it is possible, I think it would be better to add a more specific type